### PR TITLE
Improve CI with cancel & concurrency & paths filter

### PR DESCRIPTION
**Pull request by yaooqinn**
### _Why are the changes needed?_

Better ASF GHA resource usage, see more https://cwiki.apache.org/confluence/display/BUILDS/GitHub+Actions+status

- cancel-in-progress enabled. w/ this, a previous build can be canceled if new commit(s) have arrived
- style check w/o install. currently, the style check takes 8~10min to finish because an install step before it. now, it shall return within a minute. note that, this might 'fail' itself when we introduce a new module and others depend on it, but it's ok to verify by the log whether a PR ok to be merged.
- dependency check runs only when pom.xml updates. this flow now runs only pom changes as it is the only way that may cause dependency changes.

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [x] Add screenshots for manual tests if appropriate

**Canceled**: https://github.com/apache/incubator-kyuubi/runs/5604809742?check_suite_focus=true

image

- [x] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by yaooqinn** [comment]
cc @pan3793 @ulysses-you @turboFei @SteNicholas @hddong thanks.

**Review by pan3793 — APPROVED** [review]

File: .github/workflows/style.yml [hunk 1]
```diff
@@ -15,19 +15,26 @@
...
 
+concurrency:
+  group: linter-${{ github.ref }}
+  cancel-in-progress: true
+
+# no install runs before style check because it is expensive, we can skip build if style fails
+# Well, sometimes when we introduce a new module, it may 'fail' the style check for missing
+# dependency we just create for other module to inherit.
+# We can ignore this failure and merge the PR, if we see the style check pass on the new module
+  # or via a local quick verification.
```
Comment (pan3793) [inline .github/workflows/style.yml:34]: nit: indent

Comment (yaooqinn) [inline .github/workflows/style.yml:34]:
```suggestion
 # or via a local quick verification.
```

File: .github/workflows/style.yml [hunk 2]
```diff
@@ -15,19 +15,26 @@
...
 
+concurrency:
+  group: linter-${{ github.ref }}
+  cancel-in-progress: true
+
+# no install runs before style check because it is expensive, we can skip build if style fails
+# Well, sometimes when we introduce a new module, it may 'fail' the style check for missing
+# dependency we just create for other module to inherit.
+# We can ignore this failure and merge the PR, if we see the style check pass on the new module
+ # or via a local quick verification.
```
Comment (yaooqinn) [inline .github/workflows/style.yml:34]:
```suggestion
# or via a local quick verification.
```

**Comment by yaooqinn** [comment]
thanks, merged to master

**Pull request closed without merging**